llvm/lib/CodeGen/SelectionDAG/SelectionDAGBuilder.cpp

```diff
@@ -8237,6 +8237,18 @@ void SelectionDAGBuilder::visitIntrinsicCall(const CallInst &I,
     visitVectorExtractLastActive(I, Intrinsic);
     return;
   }
+  case Intrinsic::experimental_loop_dependence_war_mask:
+  case Intrinsic::experimental_loop_dependence_raw_mask: {
+    auto IntrinsicVT = EVT::getEVT(I.getType());
+    SmallVector<SDValue, 4> Ops;
+    for (auto &Op : I.operands())
+      Ops.push_back(getValue(Op));
+    unsigned ID = Intrinsic == Intrinsic::experimental_loop_dependence_war_mask
+                      ? ISD::EXPERIMENTAL_LOOP_DEPENDENCE_WAR_MASK
+                      : ISD::EXPERIMENTAL_LOOP_DEPENDENCE_RAW_MASK;
+    SDValue Mask = DAG.getNode(ID, sdl, IntrinsicVT, Ops);
+    setValue(&I, Mask);
+  }
```

Comment:
nit: The number of operands is known, so there's no need for `Ops`.
```suggestion
  case Intrinsic::experimental_loop_dependence_raw_mask:
  case Intrinsic::experimental_loop_dependence_war_mask: {
    unsigned ID = Intrinsic == Intrinsic::experimental_loop_dependence_war_mask
                      ? ISD::EXPERIMENTAL_LOOP_DEPENDENCE_WAR_MASK
                      : ISD::EXPERIMENTAL_LOOP_DEPENDENCE_RAW_MASK;
    setValue(&I, DAG.getNode(ID, sdl, EVT::getEVT(I.getType()),
                             getValue(I.getArgOperand(0)),
                             getValue(I.getArgOperand(1)),
                             getValue(I.getArgOperand(2))));
    break;
  }
```